From 51e095d8015d1e6d43385dde5b604e4777aa5fe0 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 22 Oct 2024 17:51:14 +0200
Subject: [PATCH 1451/1695] drivers: stm32_iwdg: remove registering to
 shared_resources driver

Remove registering of STM32 IWDG driver to platform shared_resources
driver that is deprecated since integration of the firewall framework
in stm32mp1 platforms. Since this integration, OP-TEE only consider
IWDG secure instances hence remove the useless code for IWDG assigned
to non-secure world.

As watchdog drivers are only used when registering to OP-TEE watchdog
services (CFG_WDT_SM_HANDLER) simplify the code to always register
IWDG instance.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b6d247b6357f4375c4bf41423de701842ec7b9d4 drivers: stm32_iwdg: remove registering to shared_resources driver]
Change-Id: If0824e33364cc937648f34b5c601fc8b19f18c36
Link: https://github.com/OP-TEE/optee_os/commit/7178041aaccf6315dda3861833ce94613cffa221
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434834
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_iwdg.c | 32 +++++---------------------------
 1 file changed, 5 insertions(+), 27 deletions(-)

diff --git a/core/drivers/stm32_iwdg.c b/core/drivers/stm32_iwdg.c
index f6e0235c4..f6a3adca8 100644
--- a/core/drivers/stm32_iwdg.c
+++ b/core/drivers/stm32_iwdg.c
@@ -75,10 +75,8 @@
 
 /*
  * Values for struct stm32_iwdg_device::flags
- * IWDG_FLAGS_NON_SECURE                Instance is assigned to non-secure world
  * IWDG_FLAGS_ENABLED			Watchdog has been enabled
  */
-#define IWDG_FLAGS_NON_SECURE			BIT(3)
 #define IWDG_FLAGS_ENABLED			BIT(4)
 
 /*
@@ -136,11 +134,6 @@ static vaddr_t get_base(struct stm32_iwdg_device *iwdg)
 	return io_pa_or_va(&iwdg->base, 1);
 }
 
-static bool is_assigned_to_nsec(struct stm32_iwdg_device *iwdg)
-{
-	return iwdg->flags & IWDG_FLAGS_NON_SECURE;
-}
-
 static void iwdg_wdt_set_enabled(struct stm32_iwdg_device *iwdg)
 {
 	iwdg->flags |= IWDG_FLAGS_ENABLED;
@@ -463,17 +456,9 @@ static TEE_Result stm32_iwdg_parse_fdt(struct stm32_iwdg_device *iwdg,
 	if (res != TEE_SUCCESS && res != TEE_ERROR_ITEM_NOT_FOUND)
 		return res;
 
-	if (dt_info.status == DT_STATUS_OK_NSEC)
-		iwdg->flags |= IWDG_FLAGS_NON_SECURE;
-
 	/* Get IOMEM address */
 	iwdg->base.pa = dt_info.reg;
-
-	if (iwdg->flags & IWDG_FLAGS_NON_SECURE)
-		io_pa_or_va_nsec(&iwdg->base, dt_info.reg_size);
-	else
-		io_pa_or_va_secure(&iwdg->base, dt_info.reg_size);
-
+	io_pa_or_va_secure(&iwdg->base, dt_info.reg_size);
 	assert(iwdg->base.va);
 
 	/* Get and check timeout value */
@@ -579,18 +564,11 @@ static TEE_Result stm32_iwdg_register(struct stm32_iwdg_device *iwdg)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
 
-	if (is_assigned_to_nsec(iwdg)) {
-		stm32mp_register_non_secure_periph_iomem(iwdg->base.pa);
-	} else {
-		stm32mp_register_secure_periph_iomem(iwdg->base.pa);
-
-		/* Expose watchdog runtime service only to secure IWDG */
-		iwdg->wdt_chip.ops = &stm32_iwdg_ops;
+	iwdg->wdt_chip.ops = &stm32_iwdg_ops;
 
-		res = watchdog_register(&iwdg->wdt_chip);
-		if (res)
-			return res;
-	}
+	res = watchdog_register(&iwdg->wdt_chip);
+	if (res)
+		return res;
 
 	SLIST_INSERT_HEAD(&iwdg_dev_list, iwdg, link);
 
-- 
2.47.2

